src/ddmd/func.d

```diff
@@ -5294,9 +5294,18 @@ extern (C++) final class InvariantDeclaration : FuncDeclaration
  */
 extern (C++) static Identifier unitTestId(Loc loc)
 {
+    // there's a counter per module in case there are two unit tests on one line
+    static __gshared int[const(char)*] counterPerModule;
```

Comment:
@klickverbot In a way that causes linker errors? I just tried a 1000-line lambda (I assume this means it won't be inlined) across modules and separate compilation was fine.

If this counter issue manifests in other ways when compiling separately, then I'm all for doing it right: it's pointless to whack one mole and leave the others. However, unless there's a clear bug elsewhere and not just with `__traits(getUnitTests)` then I think it's a big ask for me to learn the compiler internals to refactor it. Would it be nicer another way? Probably, but that's a different issue. If there are known bugs relating to this please let me know.

This patch fixes the bug, allows unittest builds with `__traits(getUnitTests)` to be built by package as is recommended and IMHO isn't entirely horrible.